### PR TITLE
[17.0][MIG] commission: Migration to 17.0

### PR DESCRIPTION
**Pull request by ambientWave**
I upgraded the 'commission' module to be installed on Odoo 17.0. More focus was given to upgrading xml files.

The demo files weren't upgraded and hence, an error during installation is warning user that demo data wasn't populated. However, the module can be used without demo data.

Python code needs to be tested thoroughly. 

All contributions and comments are welcomed.

### CODE REVIEW

**Comment by Tisho99** [comment]
Hi @ambientWave 

Thanks for the work done! 

However, the PR does not pass all the checks.

¿Are you available to finish this PR? If not, i can do it

**Comment by ambientWave** [comment]
Dear @Tisho99
The failing check is the pre-commit that does some sort of linting among other things. I'm aware of the fact that the test utilizes prettier with plugin-xml to do the work for xml files. However, there's no available information about the error that can be used as guidance to fix the issue. All I can see is that "Process completed with exit code 1." which I assume that this process ran through some kind of crashing AND "prettier (with plugin-xml)...............................................Failed" which signals that xml files are the root of the issue. Also, I'm not acquainted with knowledge about the set of rules used in the linting process of xml files. Therefore, I don't know what to do.

It would be nice if you can enlighten me on this issue. I sent you a collaboration invitation to access the repository and I appreciate your help.

**Comment by pedrobaeza** [comment]
You just need to install and run pre-commit locally:

https://github.com/OCA/maintainer-tools/wiki/Install-pre-commit

**Comment by Tisho99** [comment]
Hi @ambientWave 

As Pedro says, you just need to install and run [pre-commit](https://github.com/OCA/maintainer-tools/wiki/Install-pre-commit) locally.

It edits your code and then if you push the changes the check should be passed.

**Comment by ambientWave** [comment]
Dear @pedrobaeza
Dear @Tisho99

Thanks for your help. All checks are passed! I think PR is ready to be merged.

If there's anything that I can provide you with in the other migration PRs, just mention me.

**Comment by Tisho99** [comment]
Hi @ambientWave 

I have reviewed your PR locally and i have done a few upgrades on a copy of your branch.

https://github.com/sygel-technology/commission/commit/8cab2be8a16584f68bf8480145fa2e1d1651f62e

Can you review this changes and apply them to this branch if neccesary? You can use the git cherry pick command

It is also needed to squash the last commits into [17.0][MIG] commission: Migration to 17.0. You can use git rebase -i command, and git push -f

Once it is done, i think this PR would be ready

**Review by ambientWave — COMMENTED** [review]
I think this was a pretty good job for an important module!

**Comment by ambientWave** [comment]
Hey @Tisho99 

I'm about to apply the rebase changes. Should I use the squash or fixup command because they're similar?

The squash rebase file would look like this:

```
pick adb9c8d [MIG] commission: Migration to 17.0
squash 6af6f17 Update version to be 17.0.1.0.0 in __manifest__.py, Add newline at EOF
squash ada40ed Add newline at EOF ir.model.access.csv
squash 8fe318b [IMP] commission: pre-commit auto fixes
squash 57b3984 [IMP] commission migration

# Rebase adb9c8d..57b3984 onto adb9c8d (4 commands)
#
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup [-C | -c] <commit> = like "squash" but keep only the previous
#                    commit's log message, unless -C is used, in which case
#                    keep only this commit's message; -c is same as -C but
#                    opens the editor
# x, exec <command> = run command (the rest of the line) using shell
# b, break = stop here (continue rebase later with 'git rebase --continue')
# d, drop <commit> = remove commit
# l, label <label> = label current HEAD with a name
# t, reset <label> = reset HEAD to a label
# m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
#         create a merge commit using the original merge commit's
#         message (or the oneline, if no original merge commit was
#         specified); use -c <commit> to reword the commit message
# u, update-ref <ref> = track a placeholder for the <ref> to be updated
#                       to this position in the new commits. The <ref> is
#                       updated at the end of the rebase
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
```

**Comment by Tisho99** [comment]
Hi @ambientWave 

I only use squash if the commit messages to squash are important.

In this case you can use fixup

**Comment by ambientWave** [comment]
@Tisho99 
Commits have been squashed using fixup

File: commission/views/commission_settlement_views.xml [hunk 1]
```diff
@@ -92,7 +93,7 @@
                         <field
                             name="line_ids"
                             nolabel="1"
-                            attrs="{'readonly': [('can_edit', '=', False)]}"
+                            readonly="can_edit"
```
Comment (Tisho99) [inline commission/views/commission_settlement_views.xml]: `                            readonly="not can_edit"`

**Comment by ambientWave** [comment]
@Tisho99 Do I have to squash any modification commits from now on?

**Comment by Tisho99** [comment]
@ambientWave 

Yes, the OCA normative says that every PR should only have 1 commit to be merged. In the case of migrations, the history of commits is also added, so you have to squash the commits into "[MIG] commission: Migration to 17.0"

You can also use the git commit --amend command

Let me some time and i review again the PR

**Comment by pedrobaeza** [comment]
Let me add a bit of more information: the idea of squashing is to avoid noise having commits not adding values to the final result. Your commit `[FIX] minor fix` is just a mistake that should go in the first place into the migration commit, not isolatedly. And thus, it should be squashed together. In fact, what others do is directly to `git commit -all --amend` and `git push -f` for directly having it together, as GitHub shows the diff of the latest upload.

@Tisho99 it's allowed to have some PRs (even fixes or improvements) with several commits, if each commit is relevant enough to be isolated (for example, if you perform 2 non related improvements in the same PR).

**Review by Tisho99 — APPROVED** [review]
LGTM.  Functional and code review

**Comment by ambientWave** [comment]
@Tisho99 Squashed last commit!

**Comment by Tisho99** [comment]
Thank you! @ambientWave 

Could you do the same in the [account_commission](https://github.com/OCA/commission/pull/499) module?

File: commission/views/commission_settlement_views.xml [hunk 2]
```diff
@@ -59,15 +59,16 @@
                 <header>
                     <button
                         string="Cancel"
-                        states="settled"
+                        invisible="state not in ['settled']"
```
Comment (pedrobaeza) [inline commission/views/commission_settlement_views.xml]: Simpler:

```suggestion
                        invisible="state != 'settled'"
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please remove the file 
pandoc-3.1.11.1-windows-x86_64.msi that shouldn't be there.

File: commission/views/commission_settlement_views.xml (hunk 2 above)
Comment (ambientWave) [reply commission/views/commission_settlement_views.xml]: Dear @pedrobaeza

I applied the requested modification.

Concerning the file, It was copied to the directory by mistake and already removed from the repository.

**Comment by pedrobaeza** [comment]
I'm afraid is still not correct. In one commit, the .deb is added, and in the other, it's removed. That means twice the size of the file (35.3 MB) to download each time. Please remove the .deb in the same commit that is added, as it was never be there.

**Comment by pedrobaeza** [comment]
It seems that now, you have removed the "pre-commit fixes" commit.

**Review by pedrobaeza — APPROVED** [review]
OK, let's move with this:

/ocabot merge nobump

**Comment by ambientWave** [comment]
Dear @pedrobaeza

Here's what I have done:

1. Using rebase, I dropped the commit that included the added file which happens to be the pre-commit
2. I ran the pre-commit command
3. I commited and force pushed

**Pull request merged**